Finish off smart pointer coverage in WebGPU

geoffreygaren opened the pull request:
#### ee90c1745f1a38eb48656cbf01916274168c7984
<pre>
Finish off smart pointer coverage in WebGPU
<a href="https://bugs.webkit.org/show_bug.cgi?id=282145">https://bugs.webkit.org/show_bug.cgi?id=282145</a>
<a href="https://rdar.apple.com/138712492">rdar://138712492</a>

Reviewed by Mike Wyrzykowski.

Fixed the remaining static analyzer complaints.

I skipped false positives to do with the implementations of RefCounted,
ThreadSafeRefCounted, String, and Thread.

* Source/WebCore/Modules/WebGPU/Implementation/WebGPUQueueImpl.cpp:
(WebCore::WebGPU::QueueImpl::submit):
* Source/WebCore/Modules/WebGPU/Implementation/WebGPUQueueImpl.h:
* Source/WebCore/Modules/WebGPU/Implementation/WebGPURenderPassEncoderImpl.cpp:
(WebCore::WebGPU::RenderPassEncoderImpl::executeBundles):
* Source/WebCore/Modules/WebGPU/Implementation/WebGPURenderPassEncoderImpl.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUQueue.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPURenderPassEncoder.h:
* Source/WebGPU/WebGPU/Adapter.h:
* Source/WebGPU/WebGPU/BindGroup.h:
(WebGPU::BindGroup::bindGroupLayout const):
* Source/WebGPU/WebGPU/BindGroup.mm:
(WebGPU::BindGroup::rebindSamplersIfNeeded const):
(WebGPU::BindGroup::bindGroupLayout const): Deleted.
* Source/WebGPU/WebGPU/Buffer.h:
(WebGPU::Buffer::indirectIndexedBuffer const):
(WebGPU::Buffer::isDestroyed const):
* Source/WebGPU/WebGPU/Buffer.mm:
(WebGPU::Buffer::incrementBufferMapCount):
(WebGPU::Buffer::decrementBufferMapCount):
(WebGPU::Buffer::destroy):
(WebGPU::Buffer::isDestroyed const): Deleted.
(WebGPU::Buffer::indirectIndexedBuffer const): Deleted.
* Source/WebGPU/WebGPU/CommandEncoder.mm:
(WebGPU::CommandEncoder::discardCommandBuffer):
(WebGPU::CommandEncoder::copyBufferToTexture):
* Source/WebGPU/WebGPU/CommandsMixin.mm:
* Source/WebGPU/WebGPU/ComputePassEncoder.mm:
(WebGPU::ComputePassEncoder::executePreDispatchCommands):
(WebGPU::ComputePassEncoder::setBindGroup):
* Source/WebGPU/WebGPU/ComputePipeline.mm:
(WebGPU::ComputePipeline::getBindGroupLayout):
* Source/WebGPU/WebGPU/Device.h:
(WebGPU::Device::instance const):
(WebGPU::Device::maxBuffersPlusVertexBuffersForVertexStage const):
(WebGPU::Device::maxBuffersForFragmentStage const):
(WebGPU::Device::vertexBufferIndexForBindGroup const):
(WebGPU::Device::placeholderBuffer const):
* Source/WebGPU/WebGPU/Device.mm:
(WebGPU::Device::getXRViewSubImage):
(WebGPU::Device::makeInvalid):
(WebGPU::Device::loseTheDevice):
(WebGPU::Device::placeholderBuffer const): Deleted.
(WebGPU::Device::maxBuffersPlusVertexBuffersForVertexStage const): Deleted.
(WebGPU::Device::maxBuffersForFragmentStage const): Deleted.
(WebGPU::Device::vertexBufferIndexForBindGroup const): Deleted.
* Source/WebGPU/WebGPU/ExternalTexture.mm:
(WebGPU::ExternalTexture::destroy):
* Source/WebGPU/WebGPU/Pipeline.mm:
(WebGPU::createLibrary):
(WebGPU::errorValidatingBindGroup):
* Source/WebGPU/WebGPU/PipelineLayout.mm:
(WebGPU::PipelineLayout::errorValidatingBindGroupCompatibility const):
* Source/WebGPU/WebGPU/PresentationContextIOSurface.mm:
(WebGPU::PresentationContextIOSurface::onSubmittedWorkScheduled):
(WebGPU::PresentationContextIOSurface::getTextureAsNativeImage):
* Source/WebGPU/WebGPU/QuerySet.mm:
(WebGPU::QuerySet::destroy):
* Source/WebGPU/WebGPU/Queue.h:
* Source/WebGPU/WebGPU/Queue.mm:
(WebGPU::Queue::errorValidatingSubmit const):
(WebGPU::invalidateCommandBuffers):
(WebGPU::Queue::submit):
(WebGPU::Queue::clearTextureViewIfNeeded):
(wgpuQueueSubmit):
* Source/WebGPU/WebGPU/RenderBundle.mm:
(WebGPU::RenderBundle::replayCommands const):
(WebGPU::RenderBundle::updateMinMaxDepths):
(WebGPU::RenderBundle::validateRenderPass const):
* Source/WebGPU/WebGPU/RenderBundleEncoder.mm:
(WebGPU::RenderBundleEncoder::returnIfEncodingIsFinished):
(WebGPU::RenderBundleEncoder::executePreDrawCommands):
(WebGPU::RenderBundleEncoder::endCurrentICB):
(WebGPU::RenderBundleEncoder::finish):
(WebGPU::RenderBundleEncoder::setBindGroup):
(WebGPU::RenderBundleEncoder::setPipeline):
* Source/WebGPU/WebGPU/RenderPassEncoder.h:
(WebGPU::RenderPassEncoder::parentEncoder):
* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(WebGPU::RenderPassEncoder::RenderPassEncoder):
(WebGPU::m_maxDrawCount):
(WebGPU::RenderPassEncoder::~RenderPassEncoder):
(WebGPU::RenderPassEncoder::executePreDrawCommands):
(WebGPU::RenderPassEncoder::splitRenderPass):
(WebGPU::RenderPassEncoder::endPass):
(WebGPU::RenderPassEncoder::executeBundles):
(WebGPU::RenderPassEncoder::makeInvalid):
(WebGPU::RenderPassEncoder::setBindGroup):
(WebGPU::RenderPassEncoder::errorValidatingPipeline const):
(WebGPU::RenderPassEncoder::setPipeline):
(wgpuRenderPassEncoderExecuteBundles):
(WebGPU::RenderPassEncoder::parentEncoder): Deleted.
* Source/WebGPU/WebGPU/RenderPipeline.h:
(WebGPU::RenderPipeline::pipelineLayout const):
(WebGPU::RenderPipeline::protectedPipelineLayout const):
(WebGPU::RenderPipeline::primitiveTopology const):
(WebGPU::RenderPipeline::stripIndexFormat const):
* Source/WebGPU/WebGPU/RenderPipeline.mm:
(WebGPU::RenderPipeline::getBindGroupLayout):
(WebGPU::RenderPipeline::colorDepthStencilTargetsMatch const):
(WebGPU::RenderPipeline::pipelineLayout const): Deleted.
(WebGPU::RenderPipeline::primitiveTopology const): Deleted.
(WebGPU::RenderPipeline::stripIndexFormat const): Deleted.
* Source/WebGPU/WebGPU/ShaderModule.mm:
(WebGPU::ShaderModule::convertPipelineLayout):
* Source/WebGPU/WebGPU/Texture.h:
(WebGPU::Texture::isDestroyed const):
(WebGPU::Texture::isCanvasBacking const):
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Texture::createView):
(WebGPU::Texture::waitForCommandBufferCompletion):
(WebGPU::Texture::destroy):
(WebGPU::Texture::isCanvasBacking const): Deleted.
(WebGPU::Texture::isDestroyed const): Deleted.
* Source/WebGPU/WebGPU/TextureView.mm:
(WebGPU::TextureView::previouslyCleared const):
(WebGPU::TextureView::setPreviouslyCleared):
(WebGPU::TextureView::width const):
(WebGPU::TextureView::height const):
(WebGPU::TextureView::depthOrArrayLayers const):
(WebGPU::TextureView::destroy):
* Source/WebGPU/WebGPU/XRBinding.h:
(WebGPU::XRBinding::device):
(WebGPU::XRBinding::protectedDevice):
* Source/WebGPU/WebGPU/XRBinding.mm:
(WebGPU::XRBinding::device): Deleted.
* Source/WebGPU/WebGPU/XRProjectionLayer.mm:
(WebGPU::XRBinding::createXRProjectionLayer):
* Source/WebGPU/WebGPU/XRSubImage.mm:
(WebGPU::XRSubImage::update):
(WebGPU::XRBinding::getViewSubImage):

Canonical link: <a href="https://commits.webkit.org/285748@main">https://commits.webkit.org/285748@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ffe017634d73bc499d27bc547eec8bbfc2f270ca

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/73706 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/53135 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/26517 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/78021 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/24944 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/75823 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/62268 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/920 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/78021 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/16347 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/76773 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/48112 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/63417 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/78021 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/44837 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/20904 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/23277 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/66462 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/21252 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/79595 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/1023 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/484 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/79595 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/1165 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/63427 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/79595 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/9458 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/7639 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11361 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/987 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/3737 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/1016 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/1003 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/1022 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski reviewed (approved):
This all looks great, thank you!

mwyrzykowski reviewed (commented):
Looks like some compilation issue with `Vector<std::reference_wrapper<T>>` -> `Vector<Ref<T>>` changes

No objections to merging once that is addressed.

geoffreygaren committed:
Finish off smart pointer coverage in WebGPU
https://bugs.webkit.org/show_bug.cgi?id=282145
rdar://138712492

Reviewed by Mike Wyrzykowski.

Fixed the remaining static analyzer complaints.

I skipped false positives to do with the implementations of RefCounted,
ThreadSafeRefCounted, String, and Thread.

* Source/WebCore/Modules/WebGPU/Implementation/WebGPUQueueImpl.cpp:
(WebCore::WebGPU::QueueImpl::submit):
* Source/WebCore/Modules/WebGPU/Implementation/WebGPUQueueImpl.h:
* Source/WebCore/Modules/WebGPU/Implementation/WebGPURenderPassEncoderImpl.cpp:
(WebCore::WebGPU::RenderPassEncoderImpl::executeBundles):
* Source/WebCore/Modules/WebGPU/Implementation/WebGPURenderPassEncoderImpl.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUQueue.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPURenderPassEncoder.h:
* Source/WebGPU/WebGPU/Adapter.h:
* Source/WebGPU/WebGPU/BindGroup.h:
(WebGPU::BindGroup::bindGroupLayout const):
* Source/WebGPU/WebGPU/BindGroup.mm:
(WebGPU::BindGroup::rebindSamplersIfNeeded const):
(WebGPU::BindGroup::bindGroupLayout const): Deleted.
* Source/WebGPU/WebGPU/Buffer.h:
(WebGPU::Buffer::indirectIndexedBuffer const):
(WebGPU::Buffer::isDestroyed const):
* Source/WebGPU/WebGPU/Buffer.mm:
(WebGPU::Buffer::incrementBufferMapCount):
(WebGPU::Buffer::decrementBufferMapCount):
(WebGPU::Buffer::destroy):
(WebGPU::Buffer::isDestroyed const): Deleted.
(WebGPU::Buffer::indirectIndexedBuffer const): Deleted.
* Source/WebGPU/WebGPU/CommandEncoder.mm:
(WebGPU::CommandEncoder::discardCommandBuffer):
(WebGPU::CommandEncoder::copyBufferToTexture):
* Source/WebGPU/WebGPU/CommandsMixin.mm:
* Source/WebGPU/WebGPU/ComputePassEncoder.mm:
(WebGPU::ComputePassEncoder::executePreDispatchCommands):
(WebGPU::ComputePassEncoder::setBindGroup):
* Source/WebGPU/WebGPU/ComputePipeline.mm:
(WebGPU::ComputePipeline::getBindGroupLayout):
* Source/WebGPU/WebGPU/Device.h:
(WebGPU::Device::instance const):
(WebGPU::Device::maxBuffersPlusVertexBuffersForVertexStage const):
(WebGPU::Device::maxBuffersForFragmentStage const):
(WebGPU::Device::vertexBufferIndexForBindGroup const):
(WebGPU::Device::placeholderBuffer const):
* Source/WebGPU/WebGPU/Device.mm:
(WebGPU::Device::getXRViewSubImage):
(WebGPU::Device::makeInvalid):
(WebGPU::Device::loseTheDevice):
(WebGPU::Device::placeholderBuffer const): Deleted.
(WebGPU::Device::maxBuffersPlusVertexBuffersForVertexStage const): Deleted.
(WebGPU::Device::maxBuffersForFragmentStage const): Deleted.
(WebGPU::Device::vertexBufferIndexForBindGroup const): Deleted.
* Source/WebGPU/WebGPU/ExternalTexture.mm:
(WebGPU::ExternalTexture::destroy):
* Source/WebGPU/WebGPU/Pipeline.mm:
(WebGPU::createLibrary):
(WebGPU::errorValidatingBindGroup):
* Source/WebGPU/WebGPU/PipelineLayout.mm:
(WebGPU::PipelineLayout::errorValidatingBindGroupCompatibility const):
* Source/WebGPU/WebGPU/PresentationContextIOSurface.mm:
(WebGPU::PresentationContextIOSurface::onSubmittedWorkScheduled):
(WebGPU::PresentationContextIOSurface::getTextureAsNativeImage):
* Source/WebGPU/WebGPU/QuerySet.mm:
(WebGPU::QuerySet::destroy):
* Source/WebGPU/WebGPU/Queue.h:
* Source/WebGPU/WebGPU/Queue.mm:
(WebGPU::Queue::errorValidatingSubmit const):
(WebGPU::invalidateCommandBuffers):
(WebGPU::Queue::submit):
(WebGPU::Queue::clearTextureViewIfNeeded):
(wgpuQueueSubmit):
* Source/WebGPU/WebGPU/RenderBundle.mm:
(WebGPU::RenderBundle::replayCommands const):
(WebGPU::RenderBundle::updateMinMaxDepths):
(WebGPU::RenderBundle::validateRenderPass const):
* Source/WebGPU/WebGPU/RenderBundleEncoder.mm:
(WebGPU::RenderBundleEncoder::returnIfEncodingIsFinished):
(WebGPU::RenderBundleEncoder::executePreDrawCommands):
(WebGPU::RenderBundleEncoder::endCurrentICB):
(WebGPU::RenderBundleEncoder::finish):
(WebGPU::RenderBundleEncoder::setBindGroup):
(WebGPU::RenderBundleEncoder::setPipeline):
* Source/WebGPU/WebGPU/RenderPassEncoder.h:
(WebGPU::RenderPassEncoder::parentEncoder):
* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(WebGPU::RenderPassEncoder::RenderPassEncoder):
(WebGPU::m_maxDrawCount):
(WebGPU::RenderPassEncoder::~RenderPassEncoder):
(WebGPU::RenderPassEncoder::executePreDrawCommands):
(WebGPU::RenderPassEncoder::splitRenderPass):
(WebGPU::RenderPassEncoder::endPass):
(WebGPU::RenderPassEncoder::executeBundles):
(WebGPU::RenderPassEncoder::makeInvalid):
(WebGPU::RenderPassEncoder::setBindGroup):
(WebGPU::RenderPassEncoder::errorValidatingPipeline const):
(WebGPU::RenderPassEncoder::setPipeline):
(wgpuRenderPassEncoderExecuteBundles):
(WebGPU::RenderPassEncoder::parentEncoder): Deleted.
* Source/WebGPU/WebGPU/RenderPipeline.h:
(WebGPU::RenderPipeline::pipelineLayout const):
(WebGPU::RenderPipeline::protectedPipelineLayout const):
(WebGPU::RenderPipeline::primitiveTopology const):
(WebGPU::RenderPipeline::stripIndexFormat const):
* Source/WebGPU/WebGPU/RenderPipeline.mm:
(WebGPU::RenderPipeline::getBindGroupLayout):
(WebGPU::RenderPipeline::colorDepthStencilTargetsMatch const):
(WebGPU::RenderPipeline::pipelineLayout const): Deleted.
(WebGPU::RenderPipeline::primitiveTopology const): Deleted.
(WebGPU::RenderPipeline::stripIndexFormat const): Deleted.
* Source/WebGPU/WebGPU/ShaderModule.mm:
(WebGPU::ShaderModule::convertPipelineLayout):
* Source/WebGPU/WebGPU/Texture.h:
(WebGPU::Texture::isDestroyed const):
(WebGPU::Texture::isCanvasBacking const):
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Texture::createView):
(WebGPU::Texture::waitForCommandBufferCompletion):
(WebGPU::Texture::destroy):
(WebGPU::Texture::isCanvasBacking const): Deleted.
(WebGPU::Texture::isDestroyed const): Deleted.
* Source/WebGPU/WebGPU/TextureView.mm:
(WebGPU::TextureView::previouslyCleared const):
(WebGPU::TextureView::setPreviouslyCleared):
(WebGPU::TextureView::width const):
(WebGPU::TextureView::height const):
(WebGPU::TextureView::depthOrArrayLayers const):
(WebGPU::TextureView::destroy):
* Source/WebGPU/WebGPU/XRBinding.h:
(WebGPU::XRBinding::device):
(WebGPU::XRBinding::protectedDevice):
* Source/WebGPU/WebGPU/XRBinding.mm:
(WebGPU::XRBinding::device): Deleted.
* Source/WebGPU/WebGPU/XRProjectionLayer.mm:
(WebGPU::XRBinding::createXRProjectionLayer):
* Source/WebGPU/WebGPU/XRSubImage.mm:
(WebGPU::XRSubImage::update):
(WebGPU::XRBinding::getViewSubImage):

Canonical link: https://commits.webkit.org/285748@main
